core/services/aws/bloodDonation/completeDonationRequest.ts

```
@@ -115,6 +112,30 @@ async function completeDonationRequest(
       )
     }
 
+    await Promise.all(donorIds.map(async(donorId) => {
```

should this be a promise.allSettled? do we want to cancel everything if one fails?

---

ok, my mistake. Never used that but it should be that, thanks.

---

NO WORRIES
like if you don't need to keep the returns, you should use for loop instead; since you don't care about returns and you want to send everyone a notification! and if there is an error I assume, our listener will record these and send it to our uiLogs endpoint?

something like

```ts
for (const donorId of donorIds) {
   await sendNotification(donorId)
}
```
but if you want to send in parallel
allSettled is good but we won't be recording it.
not sure what you want to do!


you can also use a logger
and have a proxy which makes a uiLogs request and then logs the object?

just thoughts

---

I just want to send notification to all the donor in parallel and ignore the error. So I used Promise.allSettled.

---

okay, and you don't care if it errors out?

---

yes